### ISSUE TITLE
Syntax highlight breaks down for # right before an interpolated hole: $"#{foo}"

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#58637]
**Version Used**: VS2022 17.0.2

**Steps to Reproduce**:

1. Paste the following code:

```csharp
_ = $"#{42}";
```

**Expected Behavior**:

The `#` character should be colored in orange like other normal characters in the interpolated expression.

**Actual Behavior**:

The `#` character is colored in gray, like a preprocessor directive:

![image](https://user-images.githubusercontent.com/10199417/148248164-824de88a-f51e-438b-9536-dc922773dd78.png)

Builds and works just fine at runtime, it's just a visual glitch in the syntax highlight.

**Comment by CyrusNajmabadi** [comment]
Weird.  not repro-ing for me:

![image](https://user-images.githubusercontent.com/4564579/148278039-06acc8b9-3d41-4f1c-9e70-33431ce833b5.png)

**Comment by Youssef1313** [comment]
Just tried and couldn't repro as well.

**Comment by Sergio0694** [comment]
Mmmh I wonder if this might be the JetBrains Mono font, as I can see it's using a ligature there? 🤔

**Comment by jinujoseph** [comment]
cc @olegtk for info

**Comment by olegtk** [comment]
@gundermanc, @veler, this is interesting, do ligatures affect classification?

**Comment by veler** [comment]
Not AFAIK

**Comment by sharwell** [comment]
@veler We're specifically interested in the case where a ligature is formed from two characters which independently have two different classifications. For this case, either the ligature needs to be avoided or the ligature characters needs to be rendered with multiple colors.

**Comment by Sergio0694** [comment]
Ideally it would be great if the ligature could just be skipped entirely here, as having `#{` be merged makes no sense in C# anyway 🤔

**Comment by Sergio0694** [comment]
@CyrusNajmabadi @Youssef1313 Just managed to find another repro for this, happens with vanilla VS 17.4.3:

```csharp
class Repro
{
    private static int _number;

    public void Problem()
    {
        Console.WriteLine($"Hello #{_number}");
    }
}
```

Results in:

![](https://user-images.githubusercontent.com/10199417/212467243-f5dd63db-c93d-4ca6-84ea-f13b35692bac.png)

**Comment by Youssef1313** [comment]
![image](https://user-images.githubusercontent.com/31348972/212467605-10cf38d5-6c54-45c3-b5bc-55eb05f9cf03.png)

I think the font you are using is different than mine. So it may be related to the font being used.

**Comment by Sergio0694** [comment]
Yeah, I have more info. I can't repro with Consolas and Cascadia Mono. With Cascadia Code I get the same result as you (both # and { are orange). In my screenshot, I'm using JetBrains Mono. So it seem the specific font affects the syntax highlight 🤔

**Comment by Youssef1313** [comment]
I'm not sure, but this might not be a roslyn issue. More likely in VS editor (internal).

**Comment by PJB3005** [comment]
Yeah uh, something is very wrong if Roslyn even has a concept of font families in the first place.

This just looks like VS failing to break ligatures if the characters have mismatching colors, and thus rendering incorrectly.

**Comment by CyrusNajmabadi** [comment]
@olegtk to decide if you guys want to do anything here. Def not a Roslyn issue.

**Comment by olegtk** [comment]
yeah, it's a bug in the VS editor. @Sergio0694, would you mind reporting this via Help > Send Feedback > Report a Problem ? This way we will get publicly visible feedback ticket others can also vote on and comment. We would like to see more votes before we invest into fixing this, unfortunately it looks pretty complicated as we don't have much control over ligature rendering in WPF.

**Comment by CyrusNajmabadi** [comment]
Closing out as this is not in roslyn.
